File: keepr/Models/Vault.cs
namespace keepr.Models;

public class Vault : IRepoItem<int>
{
  public int Id { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
  public string Name { get; set; }
  public string Description { get; set; }
  public string CoverImg { get; set; }
  public bool IsPrivate { get; set; }
}

File: keepr.client/src/models/Vault.js
export class Vault {
  constructor(data) {
    this.id = data.id
    this.createdAt = data.createdAt
    this.updatedAt = data.updatedAt
    this.creatorId = data.creatorId
    this.creator = data.creator
    this.name = data.name
    this.description = data.description
    this.img = data.img
    this.isPrivate = data.isPrivate
  }
}